test/groovy/CloudFoundryDeployTest.groovy

```diff
@@ -0,0 +1,271 @@
+#!groovy
+import groovy.json.JsonSlurperClassic
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.ExpectedException
+import org.junit.rules.RuleChain
+import org.yaml.snakeyaml.Yaml
+import util.BasePiperTest
+import util.JenkinsEnvironmentRule
+import util.JenkinsDockerExecuteRule
+import util.JenkinsLoggingRule
+import util.JenkinsShellCallRule
+import util.JenkinsStepRule
+import util.JenkinsWriteFileRule
+import util.Rules
+
+import static org.junit.Assert.assertTrue
+import static org.junit.Assert.assertEquals
+
+class CloudFoundryDeployTest extends BasePiperTest {
+
+    private ExpectedException thrown = ExpectedException.none()
+    private JenkinsLoggingRule jlr = new JenkinsLoggingRule(this)
+    private JenkinsShellCallRule jscr = new JenkinsShellCallRule(this)
+    private JenkinsWriteFileRule jwfr = new JenkinsWriteFileRule(this)
+    private JenkinsDockerExecuteRule jedr = new JenkinsDockerExecuteRule(this)
+    private JenkinsStepRule jsr = new JenkinsStepRule(this)
+    private JenkinsEnvironmentRule jer = new JenkinsEnvironmentRule(this)
+
+    @Rule
+    public RuleChain rules = Rules
+        .getCommonRules(this)
+        .around(thrown)
+        .around(jlr)
+        .around(jscr)
+        .around(jwfr)
+        .around(jedr)
+        .around(jer)
+        .around(jsr) // needs to be activated after jedr, otherwise executeDocker is not mocked
+
+    @Before
+    void init() throws Throwable {
+        helper.registerAllowedMethod('usernamePassword', [Map], { m -> return m })
```

Comment:
In the meantime we have a [rule](https://github.com/SAP/jenkins-library/blob/master/test/groovy/util/JenkinsCredentialsRule.groovy) for that.